[WFCORE-5279] Split Elytron SSL into its own subsystem

cam-rod opened the pull request:
- https://issues.redhat.com/browse/EAP7-1564
- https://issues.redhat.com/browse/WFCORE-5279

Continuation from #371

OndrejKotek commented on elytron/WFCORE-5279-elytron-ssl-subsystem.adoc:
```diff
@@ -0,0 +1,196 @@
+authentication requirements, and also that the Elytron subsystem takes a number of resources to define a working TLS
+configuration.
+
+This RFE is to look at providing a new subsystem dedicated to a simplified TLS configuration. The Elytron subsystem
```
Do we plan to replace the default TLS configuration with this new one?

cam-rod replied:
Not at the moment, this is just planned to be a simpler way of setting up a TLS configuration. Elytron's existing code would remain for integrating with security domains, mappers, etc. when needed

OndrejKotek commented on elytron/WFCORE-5279-elytron-ssl-subsystem.adoc:
```diff
@@ -0,0 +1,196 @@
+The Undertow subsystem uses reference to an SSLContext resource provided with the 'org.wildfly.security.ssl-context' capability.
+This makes implementation of a SSL subsystem straight-forward - we need to create a   SSLContext resource with the same capability and we can wire it in by referencing its name.
+
+Unlike the 'elytron' subsystem we want the configuration as simple as possible - ideally one CLI command.
```
One command is the only simplification for users I can see in this RFE now, the rest seems be rather a code refactoring on background. Having two subsystems seems rather confusing. Are there another visible advantages? Is there a way to keep just one subsystem? (For example a Galleon layer with just TLS related configuration in the Elytron subsystem.)

cam-rod replied:
Still need to updating this section from #371. Ideally, we'd also include an updated `elytron-tool.sh` for the improvements. I'll keep this in mind as I update the section

cam-rod replied:
One thing I wanted to add from some of the earlier planning discussions is that by splitting this into it's own subsystem, it reduces the number of dependencies needed, like in non-Galleon setups, so it reduces the overall server demand.

cam-rod commented:
Updated the proposal to clarify some aspects of the scope, planned changes, and features, based on my better understanding of how components interact. Copying the commit messages below:

> **[Clarified scope, updated elytron-tls-tool](https://github.com/wildfly/wildfly-proposals/pull/500/commits/57a0bbed4080cc2f6d838318e149e96dac728995)**
> 
> Main body of the proposal has be updated to better reflect the scope
> of the subsytem (TLS only, no SASL/JASPI). Also details that the
> elytron-tls-tool should support interactive configuration, like
> add-user.

> **[Refined proposal for command and config structure](https://github.com/wildfly/wildfly-proposals/pull/500/commits/63ec10a6c1bbc9cb83688518d7de019d9164733f)**
> 
> The previous proposal discussed using both references and objects to
> store some elements (ex. key stores) within the subsystem. This makes
> it effectively incompatible with Elytron proper and also much harder to
> edit.
> 
> This updated proposal would only use "object attributes" for commands,
> but saved configuration will match that of Elytron as much as possible.

fjuma commented on elytron/WFCORE-5279-elytron-ssl-subsystem.adoc:
```diff
@@ -0,0 +1,282 @@
+In order to reduce the required configuration, dependencies, and system resources, Elytron TLS targets
+one major collection of use-cases: authentication of a server (and optionally client) using HTTPS. By limiting the
+scope to HTTP over TLS, we can remove most of the alternative authentication and authorization tools available in
+Elytron, including security domains, principal transformers, SASL support, and HTTP authentication mechanisms. This
```
Just a minor comment, I think the SASL part can be removed from here.
